src/lib/remark-plugins/remark-image-dimensions/__tests__/index.test.tsx

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import { render } from '@testing-library/react'
+import { MDXRemote } from 'next-mdx-remote'
+import { serialize } from 'next-mdx-remote/serialize'
+import Image from 'components/image'
+import { remarkPluginInjectImageDimensions, getUrlWithDimensions } from '..'
+
+const probeDimensions = { width: '500', height: '300' }
+
+// Mock the external call to get dimensions
+jest.mock('probe-image-size', () => {
+	return jest.fn(() => {
+		return probeDimensions
+	})
+})
+
+describe('remarkPluginInjectImageDimensions', () => {
+	it('does not rewrite if non-mktg-content-api url', async () => {
+		const imgSrc = `https://placehold.co/600x400/pink/FFF`
+		const alt = 'image alt'
+		const source = `
+![${alt}](${imgSrc})
+`
+		const mdxSource = await serialize(source, {
+			mdxOptions: {
+				remarkPlugins: [remarkPluginInjectImageDimensions],
+			},
+		})
+		const { getByAltText } = render(
+			<MDXRemote {...mdxSource} components={{ img: Image as any }} />
+		)
+		const img = getByAltText(alt)
+
+		expect(img).not.toHaveAttribute('width')
+		expect(img).not.toHaveAttribute('height')
+		expect(img).toHaveAttribute('src', imgSrc)
+	})
+
+	it('does not rewrite if width/ height already defined', async () => {
+		const src = `https://content.hashicorp.com/api/assets/img/themed/test-placeholder.png?width=700&height=700`
+		const url = await getUrlWithDimensions(src)
+
```

Comment:
(Just a thought): I think a helpful assertion here would be something like: 

```
expect(probe).not.toHaveBeenCalled()
```

since this is a clear indicator of "does not rewrite".

Though I wouldn't bother too much with this one as assertions on mock calls are super tedious to get correct 😅

Reply:
Ah yeah I think I tried that and it wasn't behaving as expected so I got frustrated and did this 😅 haha